### PR TITLE
feat(oidc): Yargs and email 2FA for IdP

### PR DESCRIPTION
**Pull request by atharmon**
- Uses yargs to: drop all values in all tables, run setup on the Idp to insert useful default values
- Added ability to use email as a multi-factor authorization method
- Abstracted away some of the repeated view error handling
- Set oidc-provider-nestjs port in workspace.json to port 7777 and removed other debug configs from launch.json
- Changed 2fa validation method to allow for either authenticator or totp based tokens

### CODE REVIEW

File: apps/oidc-client-test/src/main.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ async function bootstrap() {
   await app.listen(3001);
 }
 
-OpenIdClient.build(OIDC_CLIENT_METADATA_JWT, OIDC_CLIENT_PARAMS, OIDC_IDP_ISSUER_URL)
+OpenIdClient.build(OIDC_CLIENT_METADATA_BASIC, OIDC_CLIENT_PARAMS, OIDC_IDP_ISSUER_URL)
```
Comment (Proantagonist) [inline apps/oidc-client-test/src/main.ts:57]: What the difference between jwt and basic metadata?

**Review by Proantagonist — CHANGES_REQUESTED** [review]

File: apps/oidc-client-test/src/main.ts (hunk 1 above)
Comment (atharmon) [reply apps/oidc-client-test/src/main.ts:57]: It's just how the client requests an authentication method. It was always supposed to use OIDC_CLIENT_METADATA_BASIC.

Comment (atharmon) [reply apps/oidc-client-test/src/main.ts:57]: It just determines what authentication method the client wants to use. It should have stayed as OIDC_CLIENT_METADATA_BASIC.

File: apps/oidc-provider-nest/src/environments/environment.prod.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
 export const environment = {
   production: true,
   port: 4001,
-  globalPrefix: ''
+  globalPrefix: '',
+  create_defaults: true
```
[the remark this answers is not in the document]
Comment (atharmon) [reply apps/oidc-provider-nest/src/environments/environment.prod.ts:5]: Can we use [yargs](https://www.npmjs.com/package/yargs) like they use in the example?

[the remark this answers is not in the document]
Comment (Proantagonist) [reply apps/oidc-provider-nest/src/environments/environment.prod.ts:5]: Sure

**Comment by atharmon** [comment]
I'm adding some functionality to this that wasn't in the original pull request. Do we have to close this and reopen?

**Comment by Proantagonist** [comment]
Depends, what's the scope of this PR? There are no details on the description and the title is generic as hell lol. 

Don't need to close a PR if you're just adding features to a particular project. A PR is only closed when the changes are rejected, cancelled, etc. You can make another branch off of this PR's branch and add the additional features there and if this PR is not merged by the time that other PR is ready, you can make that PR dependent on this one so this PR must be merged before the other one is.

File: libs/oidc/provider-nestjs/src/lib/controllers/interaction/interaction.controller.ts [hunk 3]
```diff
@@ -143,14 +144,48 @@ export class InteractionController {
...
+      email: user.email,
+      guid: user.guid,
+      method: body.method[1],
+      error: false
+    };
+
+    if (locals.method) {
+      if (locals.method === 'totp') {
+        const token = otplib.totp.generate(user.secret2fa);
+        // Mailer.sendTokenByEmail(user, token);
```
Comment (Proantagonist) [inline libs/oidc/provider-nestjs/src/lib/controllers/interaction/interaction.controller.ts:165]: Cleanup dead code

File: libs/oidc/common/src/lib/services/user/user.service.ts [hunk 4]
```diff
@@ -44,6 +44,61 @@ export class UserService {
...
+      where: {
+        level: '99'
+      }
+    });
+
+    console.log('Admin userGuid:', entUser.guid);
+    await this.insertUserRole(entUser, adminRole, 'oidc-idp-admin');
+    // const { secretanswer1, secretanswer2, secretQuestion1, secretQuestion2 } = body;
+
+    // await this.insertSecretAnswers(secretQuestion1, secretQuestion2, secretanswer1, secretanswer2, userEnt);
```
Comment (Proantagonist) [inline libs/oidc/common/src/lib/services/user/user.service.ts:72]: Clean up dead code

File: apps/oidc-provider-nest/src/main.ts [hunk 5]
```diff
@@ -20,18 +29,81 @@ async function bootstrap() {
...
+    console.log('Not creating defaults... Loading clients');
+    const service = app.select(ClientMetadataModule).get(ClientMetadataService, { strict: true });
+
+    const clients = await service.loadClientMetadaForOidcSetup();
+    console.log('Clients:', clients);
+    OpenIdProvider.build(clients);
+  }
+
+  // Only needed during development
+  // enableOIDCDebug(OpenIdProvider.provider);
```
Comment (Proantagonist) [inline apps/oidc-provider-nest/src/main.ts:98]: Can use `environment.production` from the environments file. That way this isn't a manual comment in/out to enable it which would could also mark them as changes for future commits if one forgets to comment them out for production.

**Review by Proantagonist — CHANGES_REQUESTED** [review]
Looks pretty good. Just a couple things to clean up.

**Review by Proantagonist — APPROVED** [review]

**Pull request merged**